csrc/scheduler/transpose.cpp

```diff
@@ -90,7 +127,9 @@ class DomainMap : public pointwise_utils::DomainMap {
     return domain_map.getMappedRootDimIn(ref1, innermost2) != nullptr;
   }
 
-  size_t getInnerLeafDim(TensorView* tv, IterDomain* root_dim) const {
+  // scheduler assumes inner leaf dimension on tv is an exact mapping, when the
+  // mapping cannot be resolved, we'll return a `-1`
+  int64_t getInnerLeafDim(TensorView* tv, IterDomain* root_dim) const {
```

Comment:
Is this `root_dim` is a root domain of this `tv`? If so, is `getMappedRootDimIn` just below necessary?

Reply:
No it's not. and the mapping is necessary.
`getInnerLeafDim` is used to project the the inner leaf of another tensor onto `tv`. We could probably use a better name for this function.

i.e. in `getInnerMostDimInfoInReference`, we have vvv
```
              auto inner_most_id = scheduler_utils::innerMostRootDim(ref_tv);
              auto inner_most_pos_in_global_ref =
                  domain_map.getInnerLeafDim(global_reference, inner_most_id);
```